src/radical/pilot/agent/radical-pilot-agent-multicore.py

```diff
@@ -3873,14 +3872,18 @@ def run(self):
         #   "nc <node_ip> <port>"
         # with unique port numbers for each ExecWorker instance, obviously.
         ret, out, _  = self.launcher_shell.run_sync \
-                           ("/bin/sh %s/agent/radical-pilot-spawner.sh %s" \
-                           % (os.path.dirname (rp.__file__), self.workdir))
+                           ("/bin/sh %s/agent/radical-pilot-spawner.sh /tmp/%s-%s" \
```

Comment:
I would break out `/tmp` into a variable, and also populate that using tempfile.gettempdir().